Plugin.php

```diff
@@ -94,13 +94,18 @@ protected function forceEmailLogin(): void
         }
 
         User::extend(function ($model) {
-            $model->addDynamicMethod('getSsoId', function (string $provider) use ($model) {
-                return $model->metadata['winter.sso'][$provider]['id'] ?? null;
+            $model->addDynamicMethod('getSsoValue', function (string $provider, mixed $key, $default = null) use ($model) {
+                return $model->metadata['winter.sso'][$provider][$key] ?? $default;
             });
-            $model->addDynamicMethod('setSsoId', function (string $provider, string $id) use ($model) {
-                $metadata = $model->metadata ?? [];
-                $metadata['winter.sso'][$provider]['id'] = $id;
+            $model->addDynamicMethod('setSsoValues', function (string $provider, array $values, bool $save = false) use ($model) {
```

Comment:
Is the save parameter used right now?

Reply:
No, it's going to be used in the next PR... ;)

Reply:
Can you add it in the next PR instead then?

Reply:
I didn't catch that on the session signin_url usage and I would have preferred to have dealt with that in this PR when it gets used - makes for a cleaner history and easier to review.

Reply:
done